extensions/azure/data-plane/data-factory/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/AzureDataFactoryTransferManager.java

```diff
@@ -0,0 +1,110 @@
+
+/**
+ * Service for performing data transfers in Azure Data Factory.
+ */
+public class AzureDataFactoryTransferManager {
+    private final Monitor monitor;
+    private final Duration maxDuration;
+    private final Clock clock;
+    private final DataFactoryClient client;
+    private final DataFactoryPipelineFactory pipelineFactory;
+
+    public AzureDataFactoryTransferManager(Monitor monitor, DataFactoryClient client, DataFactoryPipelineFactory pipelineFactory, Duration maxDuration, Clock clock) {
+        this.monitor = monitor;
+        this.client = client;
+        this.pipelineFactory = pipelineFactory;
+        this.maxDuration = maxDuration;
+        this.clock = clock;
+    }
+
+    /**
+     * Transfers data from source to destination.
+     */
+    public CompletableFuture<TransferResult> transfer(DataFlowRequest request) {
+
+        PipelineResource pipeline = pipelineFactory.createPipeline(request);
+
+        var runId = client.runPipeline(pipeline).runId();
+
+        monitor.info("Created ADF pipeline for " + request.getProcessId() + ". Run id is " + runId);
+
+        return awaitRunCompletion(runId);
+    }
+
+    @NotNull
+    private CompletableFuture<TransferResult> awaitRunCompletion(String runId) {
+        monitor.info("Awaiting ADF pipeline completion for run " + runId);
+
+        var timeout = clock.instant().plus(maxDuration);
+        while (clock.instant().isBefore(timeout)) {
+            var pipelineRun = client.getPipelineRun(runId);
+            var runStatusValue = pipelineRun.status();
+            var message = pipelineRun.message();
+            monitor.info("ADF run status is " + runStatusValue + " with message [" + message + "] for run " + runId);
+            DataFactoryPipelineRunStates runStatus;
+
+            try {
+                runStatus = DataFactoryPipelineRunStates.valueOf(runStatusValue);
+            } catch (IllegalArgumentException e) {
+                return completedFuture(TransferResult.failure(ERROR_RETRY,
+                        format("ADF run in unexpected state %s with message: %s", runStatusValue, message)));
+            }
+            if (runStatus.succeeded) {
+                return completedFuture(TransferResult.success());
+            }
+            if (runStatus.failed) {
+                return completedFuture(TransferResult.failure(ERROR_RETRY,
+                        format("ADF run in state %s with message: %s", runStatusValue, message)));
+            }
+        }
+        client.cancelPipelineRun(runId);
+        return completedFuture(TransferResult.failure(ERROR_RETRY, "ADF run timed out"));
+    }
+
+    @SuppressWarnings("unused")
+    private enum DataFactoryPipelineRunStates {
+        Queued(false, false),
+        InProgress(false, false),
+        Succeeded(true, false),
+        Failed(false, true),
+        Canceling(false, true),
+        Cancelled(false, true);
```

Comment:
As I understand these statuses correspond to PipelineRun statuses from Azure SDK https://docs.microsoft.com/en-us/rest/api/datafactory/pipeline-runs/get#pipelinerun ?

Perhaps it would be helpful to add a comment about that, optionally linking to a doc

Reply:
Thanks, added comment and link